01-ruby-fundamentals/keyword-arguments.md

```diff
@@ -129,67 +129,126 @@ coffee_price(:latte, :venti, 3, false)
 coffee_price(:drip, :tall, 0, true)
 ```
 
-There are a few problems with this:
-- The caller has to remember the order of the arguments
-- When reading the call, there's no way to tell which argument is which
-- The caller must provide values for all arguments, even for the "standard" way of doing things (i.e. no extra shots)
+Having so many parameters on one method introduces several issues:
+1. The caller must remember the order of the arguments
+1. When reading the call, there's no way to tell which argument is which
+1. The caller must provide values for all arguments, even for the "standard" way of doing things (e.g. no extra shots)
 
 All of these put the onus on the caller to do things right, which is just asking for trouble. **A well-written method is easy to use right and hard to use wrong** - by that standard, this is not good code.
 
-### Keyword Arguments
+To address these issues, we will introduce two new concepts: optional arguments and keyword arguments. We'll use each by itself first, then show how they can be used together to create clean, readable code.
+
+### Optional Arguments
+
+We're going to tackle #3 on the above list first by introducing ruby's support of **Optional Arguments**.
+Also now that we're introducing a different type of argument, it's important that we recognize that the type of argument we've seen so far is generally referred to as a **Positional Argument**. The name will hopefully make more sense later on in the lesson, but for now you just need to know that's what it's called.
```

Comment:
It might make sense to move this definition of positional argument to right after line 139.

To address these issues, we will introduce two new concepts: optional arguments and keyword arguments. We'll use each by itself first, then show how they can be used together to create clean, readable code.

Now that we're introducing different types of arguments, it's important that we recognize that the type of argument we've seen so far is generally referred to as a **Positional Argument**. The name will hopefully make more sense later on in the lesson, but for now you just need to know that's what it's called.

Reply:
You're so right! That makes so much more sense! Fixing